pinot-core/src/main/java/org/apache/pinot/core/data/function/DateTimeFunctions.java

```diff
@@ -27,6 +32,8 @@
  */
 public class DateTimeFunctions {
 
+  private Map<String, DateTimeFormatter> _dateTimeFormatterMap = new HashMap<>();
```

Comment:
It seems to me that this should be a concurrent hashmap. Consumers from different partitions may call this  across threads, am I right?

Reply:
Good catch! Yeah, I'm still new to the concurrency model here - thanks for keeping an eye out for this! :)